pkg/power/platform/source/hmc.go

```diff
@@ -16,17 +13,129 @@ limitations under the License.
 
 package source
 
-type PowerHMC struct{}
+import (
+	"os"
+	"strconv"
+
+	"k8s.io/klog/v2"
+
+	"github.com/sustainable-computing-io/kepler/pkg/power/components/source"
+	"github.com/zhmcclient/golang-zhmcclient/pkg/zhmcclient"
+)
+
+const (
+	hmcSensorID string = "hmc"
+)
+
+type PowerHMC struct {
+}
+
+var hmcManager *zhmcclient.ZhmcManager
+
+func (r *PowerHMC) GetHMCManager() *zhmcclient.ZhmcManager {
+	if hmcManager == nil {
+		endpoint := os.Getenv("HMC_ENDPOINT")
+		username := os.Getenv("HMC_USERNAME")
+		password := os.Getenv("HMC_PASSWORD")
+		cacert := os.Getenv("CA_CERT")
+		skipCert := os.Getenv("SKIP_CERT")
+		isSkipCert, _ := strconv.ParseBool(skipCert)
+
+		creds := &zhmcclient.Options{Username: username, Password: password, CaCert: cacert, SkipCert: isSkipCert, Trace: false}
+		client, err := zhmcclient.NewClient(endpoint, creds, nil)
+		if err != nil {
+			klog.V(1).Infof("Error getting client connection %v", err.Message)
+		}
+		if client != nil {
+			zhmcAPI := zhmcclient.NewManagerFromClient(client)
+			hmcManager, _ := zhmcAPI.(*zhmcclient.ZhmcManager)
+			return hmcManager
+		}
+	}
+	return hmcManager
+}
+
+func (r *PowerHMC) GetEnergyFromLpar() (uint64, error) {
+	hmcManager := r.GetHMCManager()
+	lparURI := "api/logical-partitions/" + os.Getenv("LPAR_ID")
+	props := &zhmcclient.EnergyRequestPayload{
+		Range:      "last-day",
+		Resolution: "fifteen-minutes",
+	}
+	energy, _, err := hmcManager.GetEnergyDetailsforLPAR(lparURI, props)
+	if err != nil {
+		klog.V(1).Infof("Error getting energy data: %v", err.Message)
+	}
+	klog.V(1).Infof("Get energy data successfully")
+	return energy, err
+}
+
+func (r *PowerHMC) GetLiveEnergyFromLpar() (uint64, error) {
+	hmcManager := r.GetHMCManager()
+	lparURI := "/api/logical-partitions/" + os.Getenv("LPAR_ID")
+	energy, _, err := hmcManager.GetLiveEnergyDetailsforLPAR(lparURI)
+	if err != nil {
+		klog.V(1).Infof("Error getting energy data: %v", err.Message)
+	} else {
+		klog.V(1).Infof("Get energy data successfully with power: %v", energy)
+	}
+	return energy, err
+}
+
+func (r *PowerHMC) IsSystemCollectionSupported() bool {
+	return true
+}
+
+func (r *PowerHMC) StopPower() {
+}
+
+func (r *PowerHMC) GetEnergyFromDram() (uint64, error) {
+	return 0, nil
+}
+
+func (r *PowerHMC) GetEnergyFromCore() (uint64, error) {
+	return 0, nil
+}
+
+func (r *PowerHMC) GetEnergyFromUncore() (uint64, error) {
+	return 0, nil
+}
+
+func (r *PowerHMC) GetEnergyFromPackage() (uint64, error) {
+	return 0, nil
+}
+
+func (r *PowerHMC) GetNodeComponentsEnergy() map[int]source.NodeComponentsEnergy {
+	pkgEnergy, _ := r.GetLiveEnergyFromLpar()
+	pkgEnergy = pkgEnergy * 3
```

Comment:
what is '*3' for?

Reply:
It is aimed to align with `defaultSamplePeriodSec = 3` used in the kepler. Still try to finalize it and then leave explanation in the comment.